libexec/pyenv-hooks

```diff
@@ -26,7 +26,10 @@ if ! enable -f "${BASH_SOURCE%/*}"/pyenv-realpath.dylib realpath 2>/dev/null; th
     echo "pyenv: failed to load \`realpath' builtin" >&2
     exit 1
   fi
-READLINK=$(type -P greadlink readlink | head -n1)
```

Comment:
This specific order is here for a reason: in MacOS, we do prefer GNU readlink from Homebrew Coreutils to the stock BSD readlink.

But maybe this reason no longer applies?

Reply:
...and if we don't really need the extra features of GNU readlink, we can probably drop the choosing code altogether.

Reply:
> This specific order is here for a reason: in MacOS, we do prefer GNU readlink from Homebrew Coreutils to the stock BSD readlink.

If so, maybe detecting the system first would be a better choice?

> if we don't really need the extra features of GNU readlink, we can probably drop the choosing code altogether.

`$READLINK` is called without any options in `pyenv` (or so-called readlink mode). I do not know the exact difference between GNU and BSD versions of readlink, but seems in readlink mode they do basically the same thing. Maybe using either version is fine.

I searched the code and found multiple occurrences of such `readlink` detection. Then I realized the reason why Python builds incredibly slowly in WSL -- still due to failed `greadlink` search in the whole PATH before executing every command. So I believe this is a serious problem for WSL users and should be solved in some way.

Reply:
To my best knowledge, the only feature that we would need from GNU readlink is `-f`. But, since the presence of it was never a given, we have workarounds. `-f` is (sloppily) relied upon in one place in Pyenv-Virtualenv, but nowhere in Pyenv proper.

I suggest removing the chooser and see if any of the tests fail.

Reply:
We do require GNU readlink to run Bats (our test runner), there's a comment in the corresponding place. So make sure to leave the selector there. That code only runs when running tests, it doesn't affect normal use of Pyenv.

Reply:
I searched for `resolve_link` and `readlink`, seems no `-f` is used:

```
plugins/python-build/test/pyenv_ext.bats
91:resolve_link() {
243:  [[ "$(resolve_link "${INSTALL_ROOT}/bin/python")" == "python3.4" ]]
244:  [[ "$(resolve_link "${INSTALL_ROOT}/bin/python-config")" == "python3.4-config" ]]

plugins/python-build/bin/python-build
65:resolve_link() {
80:    path="$(resolve_link "$name" || true)"

libexec/pyenv
35:  resolve_link() {
50:      path="$(resolve_link "$name" || true)"

libexec/pyenv-hooks
35:resolve_link() {
47:    path="$(resolve_link "$name" || true)"

libexec/pyenv-versions
43:  resolve_link() {
56:      path="$(resolve_link "$name" || true)"
```

I will try to remove all `greadlink` names first, and wait for the CI result to fix any problem.

Reply:
Sorry to bother, but please approve the workflows @native-api

Reply:
Btw you can experiment in your fork. Just enable Github Actions in it.